_source/_posts/2022-02-28-shopping-app-with-node-minio.md

```diff
@@ -0,0 +1,709 @@
+
+This tutorial will guide you through the code to build a simple Node.js Shopping App with the Okta's Sign-In Widget for authentication and a [MinIO Server](https://min.io/) to host the store's assets. We will use Express as our application framework and Handlebars as the view engine. We will use the [MinIO Javascript Client SDK](https://docs.minio.io/docs/javascript-client-quickstart-guide) to fetch the application's image assets from the Minio Server.
+
+{% img
+blog/shopping-app-with-node-minio/home.png
+alt:"The example application"
+width:"800" %}{: .center-image }
+
+
+{% include toc.md %}
+
+**Prerequisites**
+
+This tutorial uses the following technologies but doesn't require any prior experience:
+
+* [Node.js](https://nodejs.org/en/) v16.14.0
+* [MinIO Client (mc)](https://docs.minio.io/docs/minio-client-quickstart-guide)
+* [Minio Server](https://docs.min.io/)
+* [Okta CLI](https://cli.okta.com/) v0.10.0
+
+If you'd like to skip the tutorial and just check out the fully built project, you can go [view it on GitHub](https://github.com/oktadev/okta-express-minio-app-example).
+
+## Scaffold your application
+ 
+Create a new directory for your application and navigate into it.
+
+```sh
+mkdir okta-express-minio-app-example
+cd okta-express-minio-app-example
+```
+
+Use the following command to create a `package.json` file for your application:
+
+```sh
+npm init -y
+```
+
+We need the `express` package for building our application, `express-handlebars` as the template engine and `minio` for programmatically accessing object storage. We need the `@okta/okta-signin-widget` package for authenticating users. We have added `@okta/oidc-middleware` and `express-session` so that you can continue to build this shopping app beyond the basics after this tutorial.
+
+Run the following command to install these dependencies:
+
+```sh
+npm i express@4
+npm i express-handlebars@6
+npm i express-session@1
+npm i minio@7
+npm i @okta/oidc-middleware@4
+npm i @okta/okta-signin-widget@5
+```
+
+## Set up authentication with Okta
+
+Dealing with user authentication in web apps can be a massive pain for every developer. This is where Okta shines: it helps you secure your web applications with minimal effort. 
+
+{% include setup/cli.md type="spa" loginRedirectUri="http://localhost:3000/callback" logoutRedirectUri="http://localhost:3000/" %}
+
+Open the application that you just created in the Okta Admin Dashboard. In the *General Settings* section, click on the *Edit*. Check *Interaction Code* as the *Grant Type* and click on the *Save* button.
+
+Configure the Okta tenant so that *Security* > *Profile Enrollment* > *Self-Service Registration* is enabled for your application.
+
+{% img
+blog/shopping-app-with-node-minio/SSPR.png
+alt:"SSPR"
+width:"800" %}{: .center-image }
+
+Configure the Okta tenant so that *Security* > *Authenticators* > *Password* has a rule that allows for password unlock for your application.
```

Comment:
```suggestion
Configure the Okta tenant so that *Security* > *Authenticators* > *Password* has a rule that allows password unlock for your application.
```
i don't think you need that "for" and it's smoother without...